src/neo4j_genai/components/types.py

```diff
@@ -42,44 +42,29 @@ class TextChunks(DataModel):
     chunks: list[TextChunk]
 
 
-class Neo4jProperty(BaseModel):
-    """Represents a Neo4j property.
-
-    Attributes:
-        key (str): The property name.
-        value (Any): The property value.
-    """
-
-    key: str
-    value: Any
-
-
-class Neo4jEmbeddingProperty(BaseModel):
-    """Represents a Neo4j embedding property.
-
-    Attributes:
-        key (str): The property name.
-        value (list[float]): The embedding vector.
-    """
-
-    key: str
-    value: list[float]
-
-
 class Neo4jNode(BaseModel):
     """Represents a Neo4j node.
 
     Attributes:
         id (str): The ID of the node.
         label (str): The label of the node.
-        properties (Optional[list[Neo4jProperty]]): A list of properties associated with the node.
-        embedding_properties (Optional[list[Neo4jEmbeddingProperty]]): A list of embedding properties associated with the node.
+        properties (Optional[dict[str, Any]]): A dictionary of properties attached to the node.
+        embedding_properties (Optional[dict[str, list[float]]]): A list of embedding properties attached to the node.
```

Comment:
Not for now, but I'm wondering if we could detect embeddings if they are stored in the global property list, since there must be an index in the database for them?

Reply:
Yeah maybe we could include a call to the DB which checks which properties are indexed before uploading anything. I guess the disadvantage there is that I believe you can upload the properties to the nodes then create the vector index, and that would disallow that workflow.